firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,223 @@
+        }
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        requestPermissionLauncher = null
+        cancelNotification(activity)
+    }
+
+    override fun onActivityDestroyed(activity: Activity) {
+        Log.d(TAG, "clearing current activity")
+        currentActivity = null
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        if (activity is ActivityResultCaller && !hasRequestedPermission) {
+            requestPermissionLauncher =
+                activity.registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+                    if (!isEnabled) {
+                        Log.w(
+                            TAG,
+                            "Trigger disabled after permission check. Abandoning notification."
+                        )
+                    } else if (isGranted) {
+                        showNotification(activity)
+                    } else {
+                        Log.i(TAG, "Permission not granted")
+                        // TODO: Ideally we would show a message indicating the impact of not
+                        //   enabling the permission, but there's no way to know if they've
+                        //   permanently denied the permission, and we don't want to show them a
+                        //   message after each time we try to post a notification.
+                    }
+                }
+        } else {
+            Log.w(
+                TAG,
+                "Not showing notification because this activity can't register for permission request results: $activity"
+            )
+        }
+    }
+
+    // Other lifecycle methods
+    override fun onActivityStarted(activity: Activity) {}
+    override fun onActivityStopped(activity: Activity) {}
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
+
+    fun takeScreenshot() {
+        val activity = currentActivity
+        if (activity != null) {
+            val view = activity.window.decorView.rootView
+            val bitmap = Bitmap.createBitmap(view.width, view.height, Bitmap.Config.RGB_565)
+            val canvas = Canvas(bitmap)
+            view.draw(canvas)
+            try {
+                activity.openFileOutput(SCREENSHOT_FILE_NAME, Context.MODE_PRIVATE)
+                    .use { outputStream ->
+                        bitmap.compress(
+                            Bitmap.CompressFormat.PNG, /* quality = */ 100, outputStream
+                        )
+                    }
+                Log.i(TAG, "Wrote screenshot to $SCREENSHOT_FILE_NAME")
+            } catch (e: IOException) {
+                Log.e(TAG, "Can't write $SCREENSHOT_FILE_NAME", e)
+            }
+        } else {
+            Log.e(TAG, "Can't take screenshot because current activity is unknown")
+            return
+        }
+    }
+}
+
+class TakeScreenshotAndTriggerFeedbackActivity : AppCompatActivity() {
```

Comment:
Just a note: if we're serious about supporting this use case we could add this activity to the SDK (along with tracking the previous activity so we could take the screenshot). Then they could just launch the intent for the activity and we could handle everything else. That might call for an API review though.